crates/ruma-common/src/events/poll/response.rs

```diff
@@ -53,6 +55,23 @@ impl SelectionsContentBlock {
     pub fn is_empty(&self) -> bool {
         self.0.is_empty()
     }
+
+    /// Validate these selections against the given `PollContentBlock`.
+    ///
+    /// Returns the list of valid selections in this `SelectionsContentBlock`, or `None` if there is
+    /// no valid selection.
+    pub fn validate(&self, poll: &PollContentBlock) -> Option<impl Iterator<Item = &str>> {
+        // Vote is spoiled if any answer is unknown.
+        if self.0.iter().any(|s| !poll.answers.iter().any(|a| a.id == *s)) {
+            return None;
+        }
+
+        // Fallback to the maximum value for usize because we can't have more selections than that
+        // in memory.
+        let max_selections: usize = poll.max_selections.try_into().unwrap_or(usize::MAX);
```

Comment:
I don't think this can realistically be hit with the event size limit being a thing ^^

Reply:
So should I just `unwrap` or `expect` instead?

Reply:
Ah, I misread this. Probably fine as-is.